scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,398 @@
+ *                      POST.
+ */
+dbLoader.createBulkDocsRequest = function (dataToPost, responseHandler, options) {
+    return function () {
+        var batchPostData = JSON.stringify({"docs": dataToPost});
+        options.postOptions.headers["Content-Length"] = Buffer.byteLength(batchPostData);
+        var batchDocsRequest = http.request(options.postOptions, responseHandler);
+        batchDocsRequest.write(batchPostData);
+        batchDocsRequest.end();
+        return batchDocsRequest;
+    };
+};
+
+/**
+ * Generate a response handler, setting up the given promise to resolve/reject
+ * at the correct time.
+ * @param {Function} handleEnd - Function to call that deals with the response
+ *                               data when the response receives an "end" event.
+ * @param {Promise} promise - Promose to resolve/reject on a response "end" or
+ *                           "error" event.
+ * @param {String} errorMsg - Optional error message to prepend to the error
+ *                            received from a response "error" event.
+ * @return {Function} - Function reponse callback for an http request.
+ */
+dbLoader.createResponseHandler = function (handleEnd, promise, errorMsg) {
+    return function (response) {
+        var responseString = "";
+
+        response.setEncoding("utf8");
+        response.on("data", function (chunk) {
+            responseString += chunk;
+        });
+        response.on("end", function () {
+            if (response.statusCode >= 400) {   // error
+                var fullErrorMsg = errorMsg +
+                                   response.statusCode + " - " +
+                                   response.statusMessage;
+                // Document-not-found or 404 errors include a reason in the
+                // response.
+                // http://docs.couchdb.org/en/stable/api/basics.html#http-status-codes
+                if (response.statusCode === 404) {
+                    fullErrorMsg = fullErrorMsg + ", " +
+                                   JSON.parse(responseString).reason;
+                }
+                promise.reject(fullErrorMsg);
+            }
+            else {
+                var value = handleEnd(responseString);
+                promise.resolve(value);
+            }
+        });
+        response.on("error", function (e) {
+            fluid.log(errorMsg + e.message);
+            promise.reject(e);
+        });
+    };
+};
+
+/**
+ * Quit the whole process because a request of the database has failed, and log
+ * the error.  Use this function when the failure has not actually modified the
+ * database; for example, when getting all the current snapset Prefs Safes.  If
+ * that failed, that database is as it was, but there is no point in continuing.
+ * @param {String} errorMsg - The reason why database access failed.
+ */
+dbLoader.bail = function (errorMsg) {
+    fluid.log(errorMsg);
+    process.exit(1);
```

Comment:
This isn't an appropriate place to write code which destroys the whole process. Instead push this into the top-level orchestrator's rejection handler, if necessary - but I believe it won't be. Any rejection will terminate the whole activity by default, if you use fluid.promise.sequence.

Reply:
I think I've got it.  After reworking `orchestrate()` to use `fluid.promise.sequence()`, I've puat the exit code in the promise returned by the sequence, [see line 517](https://github.com/GPII/universal/blob/230c8f078f508ad3189a460d9567840db151150c/scripts/deleteAndLoadSnapsets.js#L517).  Is that what you mean?